### PR TITLE
fix(healthcare): replace deprecated NotificationConfig with FhirNotificationConfig

### PR DESCRIPTION
**Pull request by noerog**
## Description

Fixes b/310269530

## Checklist
- [X] I have followed [Contributing Guidelines from CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md)
- [X] **Tests** pass:   `go test -v ./..` (see [Testing](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#testing))
- [X] **Code formatted**:   `gofmt` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [X] **Vetting** pass:   `go vet` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [X] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: healthcare/fhir_store_patch.go [hunk 1]
```diff
@@ -36,11 +36,11 @@ func patchFHIRStore(w io.Writer, projectID, location, datasetID, fhirStoreID, to
...
-	if _, err := storesService.Patch(name, &healthcare.FhirStore{
-		NotificationConfig: &healthcare.NotificationConfig{
-			PubsubTopic: topicName, // format is "projects/*/locations/*/topics/*"
-		},
-	}).UpdateMask("notificationConfig").Do(); err != nil {
+	notificationConfig := &healthcare.FhirNotificationConfig{
+		PubsubTopic: topicName, // format is "projects/*/locations/*/topics/*"
+	}
+
+	if _, err := storesService.Patch(name, &healthcare.FhirStore{NotificationConfigs: []*healthcare.FhirNotificationConfig{notificationConfig}}).UpdateMask("notificationConfigs").Do(); err != nil {
```
Comment (telpirion) [inline healthcare/fhir_store_patch.go:43]: issue: initialize the `healthcare.FhirStore` object and call `UpdateMask()` outside of this block.

This is difficult to read as written.

File: healthcare/fhir_store_patch.go [hunk 2]
```diff
@@ -36,11 +36,11 @@ func patchFHIRStore(w io.Writer, projectID, location, datasetID, fhirStoreID, to
 
 	name := fmt.Sprintf("projects/%s/locations/%s/datasets/%s/fhirStores/%s", projectID, location, datasetID, fhirStoreID)
 
-	if _, err := storesService.Patch(name, &healthcare.FhirStore{
-		NotificationConfig: &healthcare.NotificationConfig{
-			PubsubTopic: topicName, // format is "projects/*/locations/*/topics/*"
-		},
-	}).UpdateMask("notificationConfig").Do(); err != nil {
+	notificationConfig := &healthcare.FhirNotificationConfig{
+		PubsubTopic: topicName, // format is "projects/*/locations/*/topics/*"
```
Comment (telpirion) [inline healthcare/fhir_store_patch.go:40]: nit: I would prefer to have this comment above the field. Up to you!

**Review by telpirion — CHANGES_REQUESTED** [review]

File: healthcare/fhir_store_patch.go (hunk 2 above)
Comment (noerog) [reply healthcare/fhir_store_patch.go:40]: Done.

File: healthcare/fhir_store_patch.go (hunk 1 above)
Comment (noerog) [reply healthcare/fhir_store_patch.go]: Done.

**Review by telpirion — APPROVED** [review]
Thank you very much!

**Pull request merged**